stdlib/public/core/Collection.swift

```diff
@@ -260,9 +260,10 @@ extension CollectionType {
 
   /// Returns the first element of `self`, or `nil` if `self` is empty.
   ///
-  /// - Complexity: O(1)
+  /// - Complexity: O(1) in most cases, possibly O(N) for lazy collections.
   public var first: Generator.Element? {
-    return isEmpty ? nil : self[startIndex]
+    var gen = generate()
+    return gen.next()
```

Comment:
I believe I ran the regular test suite but not the validation suite. I'd check but I'm not at my computer right now. If you don't merge it tonight then I'll make sure to re-run it tomorrow just to be sure.